myaws/autoscaling_set_instance_protection.go

```diff
@@ -0,0 +1,37 @@
+package myaws
+
+import (
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/pkg/errors"
+)
+
+// AutoScalingSetInstanceProtectionOptions customize the behavior of the Attach command.
+type AutoScalingSetInstanceProtectionOptions struct {
+	AsgName              string
+	InstanceIds          []*string
+	ProtectedFromScaleIn bool
+}
+
+// AutoScalingSetInstanceProtection protects from termination when scale in your autoscaling group.
+func (client *Client) AutoScalingSetInstanceProtection(options AutoScalingSetInstanceProtectionOptions) error {
+	if len(options.InstanceIds) > 0 {
+		if err := client.AutoScalingSetInstanceProtectionInstances(options.AsgName, options.InstanceIds, options.ProtectedFromScaleIn); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (client *Client) AutoScalingSetInstanceProtectionInstances(asgName string, instanceIds []*string, protectedFromScaleIn bool) error {
+	params := &autoscaling.SetInstanceProtectionInput{
+		AutoScalingGroupName: &asgName,
+		InstanceIds:          instanceIds,
```

Comment:
According to the API doc, we can only set up to 19 instances.
If your use-case does not exceed this limit, pagination is not required for now, but at least add the limit to comments.
https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_SetInstanceProtection.html

Reply:
Thank you.
I fixed code to process every 19 instances.